### PR TITLE
Add documentation on modularized state.

### PR DESCRIPTION
**Pull request by sgomes**
#### Changes proposed in this Pull Request

* Update existing documentation with modularized state info

#### Testing instructions

Documentation-only changes; no need for testing.

### CODE REVIEW

File: client/state/README.md [hunk 1]
```diff
@@ -3,7 +3,36 @@ State
...
+function getStream( state, streamKey ) {
+	return state.reader.streams[ streamKey ] || emptyStream;
+}
+```
+
+This way, registration happens automatically as part of resolving the dependency graph, and state gets distributed across chunks through the regular build process, without any manual input.
+
+In order to centralize state, it's recommended that all reducers, selectors and action creators for a given portion of state live under the corresponding directory in `client/state/<name>`. Selectors that access multiple parts of state should live under `client/state/selectors`.
+
+There is currently an ongoing migration effort in modularizing all of Calypso state, following the approach above. Over time, the number of statically loaded reducers in `index.js` will go down, until we reach an empty set and handle all state loading through the modularized approach.
```
Comment (tyxla) [inline client/state/README.md]: I'm not sure we'll eventually reach an "empty set" 😉 . I'd imagine that there is a crucial set of minimal reducer subtrees that will always be there and will have to be loaded with the main app - for example `sites`, `currentUser`, `ui`, etc. What do you think?

**Review by tyxla — APPROVED** [review]
Love the changes, thank you ❤️ ! This is ready to ship IMHO, I just asked a question that doesn't block the PR by any means.

File: client/state/README.md (hunk 1 above)
Comment (sgomes) [reply client/state/README.md]: Those examples would most likely be there on every route, yes, but they can still be loaded through the same modularised mechanism as the rest. So `index.js` would contain an empty set, but every route would still load *some* set of essential reducers.

The cool thing is that we wouldn't even have to determine what this set of essential reducers would be; the modularised mechanism would sort it for us, automatically 🙂

Comment (tyxla) [reply client/state/README.md]: I'm excited to get closer to that future, Sérgio! That's pretty cool 💯 ❤️

Comment (tyxla) [reply client/state/README.md]: That being said, I'm happy to 🚢 this 😉

**Comment by tyxla** [comment]
@sgomes I'd also say this selectors nuance is definitely p2-worthy material.

File: client/state/README.md (hunk 1 above)
Comment (sgomes) [reply client/state/README.md]: > I'm excited to get closer to that future, Sérgio! That's pretty cool 💯 ❤️

Very much looking forward to that too! 😄

**Comment by sgomes** [comment]
Thank you again for the review, @tyxla! 👍 I'll plan a p2 post for the near future.

**Pull request merged**